core/src/main/scala/org/apache/spark/ui/exec/ExecutorsTab.scala

```diff
@@ -70,15 +72,33 @@ class ExecutorsListener(storageStatusListener: StorageStatusListener, conf: Spar
     executorToLogUrls(eid) = executorAdded.executorInfo.logUrlMap
     executorToTotalCores(eid) = executorAdded.executorInfo.totalCores
     executorToTasksMax(eid) = executorToTotalCores(eid) / conf.getInt("spark.task.cpus", 1)
-    executorIdToData(eid) = new ExecutorUIData(executorAdded.time)
+    executorEvents += executorAdded
+    if (executorEvents.size > MAX_EXECUTOR_LIMIT) {
+      executorEvents = executorEvents.drop(1)
```

Comment:
Because drop function don't really remove element, it just return a new collection.